src/hooks/index.ts

```diff
@@ -164,14 +166,46 @@ export function useEagerConnect() {
   const [tried, setTried] = useState(false)
   const [isManuallyDisconnect] = useIsUserManuallyDisconnect()
   const [isAcceptedTerm] = useIsAcceptedTerm()
-
+  const detectedWallet = detectInjectedType()
+  const [, setConnectedWallet] = useConnectedWallet()
   useEffect(() => {
-    try {
-      // If not accepted Terms or Terms changed: block eager connect for EVM wallets and disconnect manually for Solana wallet
-      if (!isAcceptedTerm) {
+    const func = async () => {
+      try {
+        // If not accepted Terms or Terms changed: block eager connect for EVM wallets and disconnect manually for Solana wallet
+        if (!isAcceptedTerm) {
+          setTried(true)
+          disconnect()
+          setConnectedWallet(undefined)
+          return
+        }
+        const authorized = await isAuthorized()
         setTried(true)
-        disconnect()
-        return
+        let connector,
+          throwError = true
+        // try to connect if previous connected to Coinbase Link
+        if (authorized && window.localStorage.getItem(WALLETLINK_LOCALSTORAGE_NAME) && !isManuallyDisconnect) {
+          connector = walletlink
+          throwError = false
+        } else if ((authorized && !isManuallyDisconnect) || (isMobile && window.ethereum)) {
+          const lastWalletKey = localStorage.getItem(LOCALSTORAGE_LAST_WALLETKEY)
+          const wallet = lastWalletKey && SUPPORTED_WALLETS[lastWalletKey]
+          connector = wallet && isEVMWallet(wallet) ? wallet.connector : injected
+        } else if (detectedWallet === 'TRUST_WALLET' && !isManuallyDisconnect) {
+          connector = trustWalletConnector
+        } else if (detectedWallet === 'WALLET_CONNECT' && !isManuallyDisconnect) {
+          connector = walletconnect
+        }
+        if (connector) {
+          await activate(connector, undefined, throwError)
+          const account = await connector.getAccount()
+          setConnectedWallet(account || undefined)
+        } else {
+          setConnectedWallet(undefined)
+        }
+      } catch (e) {
+        console.log('Eagerly connect: authorize error', e)
+        setTried(true)
+        setConnectedWallet(undefined)
       }
```

Comment:
Is this an optional additional improvement? I believe this is not in the scope of KyberAI. I'm refactoring this in PR #1954 (update web3 v8) and this will likely be unused in new web3 v8.

Reply:
same as https://github.com/KyberNetwork/kyberswap-interface/pull/1577#discussion_r1214409812
<img width="909" alt="image" src="https://github.com/KyberNetwork/kyberswap-interface/assets/33005392/c491213b-1025-4c0e-bd86-e57d748814cf">